connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -82,11 +93,19 @@
         Arrays.stream(SERVICE_LOADER_PLUGINS).map(serviceLoaderPlugin -> MANIFEST_PREFIX + serviceLoaderPlugin.getName())
             .collect(Collectors.toSet());
 
+    // Although this classloader does not load classes directly but rather delegates loading to a
+    // PluginClassLoader or its parent through its base class, because of the use of inheritance in
+    // in the latter case, this classloader needs to also be declared as parallel capable to use
+    // fine-grain locking when loading classes.
+    static {
```

Comment:
I think it would be clearer to put this with the static fields rather than here after the member fields.

Also, is it worth mentioning that this CL is parallel-capable because it's superclass is, and we haven't overridden loadClass/defineClass, and so no additional sychronization is necessary?

Reply:
Keeping the static block here, because it's a block and that's what we have in `PluginClassLoader`. Our style is not too strict with respect of this ordering. 

The class overwrites the `loadClass` method. But it's delegating the loading to different classloaders and the locking is embedded in these classes.